src/typescript/fable-splitter/index.ts

```diff
@@ -0,0 +1,205 @@
+import * as Babel from "babel-core";
+import * as fs from "fs";
+import * as Path from "path";
+import * as Process from "process";
+const client = require("fable-utils/client");
+const babelPlugins = require("fable-utils/babel-plugins");
+
+const customPlugins = [
+    babelPlugins.getRemoveUnneededNulls(),
+    babelPlugins.getTransformMacroExpressions(Babel.template),
+];
+
+const DEFAULT_PORT = parseInt(process.env.FABLE_SERVER_PORT || "61225", 10);
+const FSHARP_EXT = /\.(fs|fsx|fsproj)$/;
+const FSPROJ_EXT = /\.fsproj$/;
+const JAVASCRIPT_EXT = /\.js$/;
+
+let alreadyCompiled: Set<string>;
+
+export type FableOptions = {
+    path?: string;
+    define?: string[];
+    plugins?: string[];
+    fableCore?: string;
+    declaration?: boolean;
+    typedArrays?: boolean;
+    clampByteArrays?: boolean;
+    extra?: any;
+};
+
+export type FableCompilerOptions = {
+    entry: string;
+    outDir: string;
+    port?: number;
+    babel?: Babel.TransformOptions;
+    fable?: FableOptions;
+};
+
+function output(key: string) {
+    switch (key) {
+        case "warning": return console.warn;
+        case "error": return console.error;
+        default: return console.log;
+    }
+}
+
+function ensureArray(obj: any) {
+    return (Array.isArray(obj) ? obj : obj != null ? [obj] : []);
+}
+
+function ensureDirExists(dir: string) {
+    if (dir && /(?:\/|\\)/.test(dir)) {
+        ensureDirExists(Path.dirname(dir));
+    }
+    if (!fs.existsSync(dir)) {
+        fs.mkdirSync(dir);
+    }
+}
+
+// flat folder structure (one level deep)
+function flatPath(fullPath: string) {
+    const fileName = Path.basename(fullPath)
+        .replace(FSHARP_EXT, "").replace(JAVASCRIPT_EXT, "");
+    const fileDir = Path.basename(Path.dirname(fullPath));
+    const newPath = Path.join(fileDir, fileName);
+    return newPath;
+}
+
+function fixPath(dir: string, path: string) {
+    if (!path.startsWith(".")) { return path; } // node package
+    const relPath = Path.join(dir, path);
+    const fullPath = Path.resolve(relPath);
+    const newPath = Path.join("..", flatPath(fullPath));
+    return newPath.replace(/\\/g, "/");
```

Comment:
@alfonsogarciacaro Fixing the extension is done in flatPath. I'll add another cache there for name dedup, since the alreadyCompiled cache is for checking the source path.